airbyte-integrations/connectors/source-dockerhub/source_dockerhub/source.py

```diff
@@ -0,0 +1,88 @@
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.requests_native_auth import TokenAuthenticator
+from urllib.parse import urlparse
+
+class SourceDockerhub(AbstractSource):
+    jwt = None
+
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        username = config["docker_username"]
+
+        # get JWT
+        jwt_url = "https://auth.docker.io/token?service=registry.docker.io&scope=repository:library/alpine:pull"
+        response = requests.get(jwt_url)
+        self.jwt = response.json()["token"]
+
+
+        # check that jwt is valid and that username is valid
+        url = f"https://hub.docker.com/v2/repositories/{username}/"
+        auth = TokenAuthenticator(token=self.jwt)
+        try:
+            # response = requests.get(url, headers=auth.get_auth_header())
+            response = requests.get(url, headers={"Authorization": self.jwt})
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == 401:
+                print(str(e))
+                return False, "Invalid JWT received, check if auth.docker.io changed API"
+            elif e.response.status_code == 404:
+                print(str(e))
+                return False, f"User '{username}' not found, check if hub.docker.com/u/{username} exists"
+            else:
+                print(str(e))
+                return False, f"Error getting basic user info for Docker user '{username}', unexpected error"
+        json_response = response.json()
+        repocount = json_response["count"]
+        print(f"Connection check for Docker user '{username}' successful: {repocount} repos found")
+        return True, None
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        return [DockerHub(jwt=self.jwt, config=config)]
+
+class DockerHub(HttpStream):
+    url_base = "https://hub.docker.com/v2"
+
+    # Set this as a noop.
+    primary_key = None
+
+    def __init__(self, jwt: str, config: Mapping[str, Any], **kwargs):
+        super().__init__(**kwargs)
```

Comment:
nit: Since you're not actually passing any `**kwargs` in when you instantiate this you can just call super like `super().__init__()`